components/script/dom/filereadersync.rs

```diff
@@ -31,3 +45,108 @@ impl FileReaderSync {
         Ok(FileReaderSync::new(global))
     }
 }
+
+impl FileReaderSyncMethods for FileReaderSync {
+    // https://w3c.github.io/FileAPI/#readAsBinaryStringSyncSection
+    fn ReadAsBinaryString(&self, blob: &Blob) -> Fallible<DOMString>{
+        // step 1
+        if self.ready_state.get() == FileReaderReadyState::Loading {
+            return Err(Error::InvalidState);
+        }
+
+        // step 2
+        let blob_contents = blob.get_bytes().unwrap_or(vec![]);
+        if blob_contents.is_empty() {
```

Comment:
What happens if we run a test like `readerSync.readAsBinaryString(new Blob([""]));`? Won't this incorrectly throw an exception?